tools/mo/openvino/tools/mo/front/mxnet/arange_like_replacer.py

```diff
@@ -0,0 +1,79 @@
+from openvino.tools.mo.ops.shape import Shape
+from openvino.tools.mo.ops.squeeze import Squeeze
+from openvino.tools.mo.utils.error import Error
+
+
+class ArangeLikeReplacer(FrontReplacementOp):
+    op = 'arange_like'
+    enabled = True
+
+    def replace_sub_graph(self, graph: Graph, match: dict):
+        node = match['op']
+        name = node.soft_get('name', node.id)
+        if node.repeat != 1:
+            raise Error("arange_like node {} with non default repeat value {} "
+                        "is not supported".format(name, node.repeat))
+        axis = node.axis
+        shape_node = Shape(graph, {'name': name + '/shape'}).create_node()
+        range_node = create_op_with_const_inputs(graph, Range, {0: int64_array(node.start),
+                                                                2: int64_array(node.step)}, {'name': name + '/Range'})
+        node.in_port(0).get_connection().set_destination(shape_node.in_port(0))
+
+        if axis is not None:
+            '''
+            Replace arange_like op to subgraph:
+            Shape - Gather - Range
+            '''
+            gather_node = create_op_with_const_inputs(graph, Gather, {1: int64_array([axis]),
+                                                                      2: int64_array(0)},
+                                                      {'name': name + '/Gather'})
+            shape_node.out_port(0).connect(gather_node.in_port(0))
+            gather_node.out_port(0).connect(range_node.in_port(1))
+            node.out_port(0).get_connection().set_source(range_node.out_port(0))
+            rename_nodes([(node, name + '/ShouldBeDeleted'), (range_node, name)])
+        else:
+            r'''
+            Replace arange_like op to subgraph:
+                    |
+                  Shape     
+                 /     \
+            Reshape     |
+               |        |
+             Range      | 
+                \      /
+                 Reshape
+                    |
+            '''
+
+            reshape_node1 = create_op_with_const_inputs(graph, Reshape, {1: int64_array([-1])},
+                                                        {'name': name + '/Reshape_forward'})
+
+            reshape_node2 = Reshape(graph, {'name': name + '/Reshape_backward'}).create_node()
+            shape_node.out_port(0).connect(reshape_node1.in_port(0))
+            reshape_node1.out_port(0).connect(range_node.in_port(1))
+            range_node.out_port(0).connect(reshape_node2.in_port(0))
+            shape_node.out_port(0).connect(reshape_node2.in_port(1))
+            node.out_port(0).get_connection().set_source(reshape_node2.out_port(0))
+            rename_nodes([(node, name + '/ShouldBeDeleted'), (reshape_node2, name)])
+
+        # MXNet arange_like op has no stop attribute and the result tensor always matches the input shape, so
+        # we have to correct the stop value for the Range node if start > 0
+        if node.start > 0:
```

Comment:
minor: start can't be negative, right? Should we check it somewhere (assert)?

Reply:
I guess, there should be `if node.start != 0`